source/common/grpc/status.cc

```diff
@@ -85,5 +88,34 @@ uint64_t Utility::grpcToHttpStatus(Status::GrpcStatus grpc_status) {
   }
 }
 
+absl::optional<Status::GrpcStatus> Utility::nameToGrpcStatus(const std::string& grpc_status_name) {
+  const static std::unordered_map<std::string, Status::GrpcStatus> status_map({
+      {"ok", Status::GrpcStatus::Ok},
+      {"canceled", Status::GrpcStatus::Canceled},
+      {"unknown", Status::GrpcStatus::Unknown},
+      {"invalid_argument", Status::GrpcStatus::InvalidArgument},
+      {"deadline_exceeded", Status::GrpcStatus::DeadlineExceeded},
+      {"not_found", Status::GrpcStatus::NotFound},
+      {"already_exists", Status::GrpcStatus::AlreadyExists},
+      {"permission_denied", Status::GrpcStatus::PermissionDenied},
+      {"resource_exhausted", Status::GrpcStatus::ResourceExhausted},
+      {"failed_precondition", Status::GrpcStatus::FailedPrecondition},
+      {"aborted", Status::GrpcStatus::Aborted},
+      {"out_of_range", Status::GrpcStatus::OutOfRange},
+      {"unimplemented", Status::GrpcStatus::Unimplemented},
+      {"internal", Status::GrpcStatus::Internal},
+      {"unavailable", Status::GrpcStatus::Unavailable},
+      {"data_loss", Status::GrpcStatus::DataLoss},
+      {"unauthenticated", Status::GrpcStatus::Unauthenticated},
```

Comment:
It'd be good to add a comment to ensure that this list and the [configuration listing](https://github.com/envoyproxy/envoy/pull/5682/files#diff-329f6ccdc5e5ba4ada8f91cb0b5b7407R199) is updated if there are changes to the `GrpcStatus` code `enum` at:

https://github.com/envoyproxy/envoy/blob/dfbb069d634884e386130dc1b74ef0ff2d4657b0/include/envoy/grpc/status.h#L8.